packages/block/test/eip7685block.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { Chain, Common, Hardfork } from '@ethereumjs/common'
+import { CLRequest, KECCAK256_RLP, concatBytes, hexToBytes, randomBytes } from '@ethereumjs/util'
+import { assert, describe, expect, it } from 'vitest'
+
+import { Block, BlockHeader } from '../src/index.js'
+
+import type { CLRequestType } from '@ethereumjs/util'
+
+class NumberRequest extends CLRequest implements CLRequestType {
+  constructor(type: number, bytes: Uint8Array) {
+    super(type, bytes)
+  }
+
+  public static fromRequestData(bytes: Uint8Array): CLRequestType {
+    return new NumberRequest(0x1, bytes)
+  }
+
+  serialize() {
+    return concatBytes(Uint8Array.from([this.type]), this.bytes)
+  }
+}
+
+const common = new Common({
+  chain: Chain.Mainnet,
+  hardfork: Hardfork.Cancun,
+  eips: [7685, 4844, 4788],
+})
+describe('7685 tests', () => {
+  it('should instantiate block with defaults', () => {
+    const block = Block.fromBlockData({}, { common })
+    assert.deepEqual(block.header.requestsRoot, KECCAK256_RLP)
+    const block2 = new Block(undefined, undefined, undefined, undefined, { common })
+    assert.deepEqual(block.header.requestsRoot, KECCAK256_RLP)
+    assert.equal(block2.requests?.length, 0)
+  })
+  it('should instantiate a block with requests', async () => {
+    const request = new NumberRequest(0x1, randomBytes(32))
+    const requestsRoot = await Block.genRequestsTrieRoot([request])
+    const block = Block.fromBlockData(
+      {
+        requests: [request],
+        header: { requestsRoot },
+      },
+      { common }
+    )
+    assert.equal(block.requests?.length, 1)
+    assert.deepEqual(block.header.requestsRoot, requestsRoot)
+  })
+  it('RequestsRootIsValid should return false when requestsRoot is invalid', async () => {
+    const request = new NumberRequest(0x1, randomBytes(32))
+    const block = Block.fromBlockData(
+      {
+        requests: [request],
+        header: { requestsRoot: randomBytes(32) },
+      },
+      { common }
+    )
+
+    assert.equal(await block.requestsTrieIsValid(), false)
+  })
+  it('should validate requests order', async () => {
+    const request1 = new NumberRequest(0x1, hexToBytes('0x1234'))
+    const request2 = new NumberRequest(0x1, hexToBytes('0x2345'))
+    const request3 = new NumberRequest(0x2, hexToBytes('0x2345'))
+    const requests = [request1, request2, request3]
+    const requestsRoot = await Block.genRequestsTrieRoot(requests)
+
+    // Construct block with requests in correct order
+
+    const block = Block.fromBlockData(
+      {
+        requests,
+        header: { requestsRoot },
+      },
+      { common }
+    )
+
+    assert.ok(await block.requestsTrieIsValid())
+
+    // Throws when requests are not ordered correctly
+    await expect(async () =>
+      Block.fromBlockData(
+        {
+          requests: [request1, request3, request2],
+          header: { requestsRoot },
```

Comment:
For completeness maybe also calculate the correct `requestsRoot` for this requests array? (I am assuming that the `toThrow` thing checks for the correct error message, though?)

Reply:
This is happening in line 66 above (assuming the requests are in the correct order). There is no valid `requestsRoot` for a requests array that isn't in the correct order.

The `toThrow` check examines the error message to confirm that the error is regarding the order of the requests (i.e. throws if the request array isn't monotonically increasing)

Reply:
Ok, since `toThrow` checks for the correct error this is fine :smile: